src/String.js

```diff
@@ -79,11 +84,29 @@ class StringT extends Base {
     stream.writeString(val, encoding);
 
     if ((this.length == null)) {
-      return stream.writeUInt8(0x00);
+      return encodingWidth(encoding) == 2 ?
```

Comment:
In `encodingWidth` we check for more values, but here we ignore at least `utf16be` and write it as `utf16le`.

Reply:
yes, but it seems strange to write `utf16be` with `stream.writeUInt16LE`.

Reply:
I guess, but endianness is not terribly relevant when writing zeros!